blaze-client/src/main/scala/org/http4s/client/blaze/Http1Connection.scala

```diff
@@ -179,11 +179,11 @@ private final class Http1Connection(val requestKey: RequestKey,
         val httpVersion = parser.getHttpVersion()
 
         // we are now to the body
-        def terminationCondition() = stageState.get match {  // if we don't have a length, EOF signals the end of the body.
-          case Error(e) if e != EOF => e
+        def terminationCondition(): Either[Throwable, Option[Chunk[Byte]]] = stageState.get match {  // if we don't have a length, EOF signals the end of the body.
+          case Error(e) if e != EOF => Left(e)
           case _ =>
-            if (parser.definedContentLength() || parser.isChunked()) InvalidBodyException("Received premature EOF.")
-            else Terminated(End)
+            if (parser.definedContentLength() || parser.isChunked()) Left(InvalidBodyException("Received premature EOF."))
+            else Right(None)
```

Comment:
Using `Either.right(None)` and `Either.left(InvalidBodyException ...)` will allow you to get rid of explicit return type annotation (after importing `cats.syntax.either._`).